src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -393,4 +395,49 @@ private boolean userHasPermission(Course course) {
         courseService.delete(id);
         return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(ENTITY_NAME, title)).build();
     }
+
+    /**
+     * GET  /courses/:courseId/results : Returns all results of the exercises of a course for the currently logged in user
+     *
+     * @param courseId the id of the course to get the results from
+     * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
+     */
+    @GetMapping(value = "/courses/{courseId}/results")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @Timed
+    @Transactional(readOnly = true)
+    public ResponseEntity<Course> getResultsForCurrentStudent(@PathVariable Long courseId) {
+        long start = System.currentTimeMillis();
+        log.debug("REST request to get Results for Course and current Studen : {}", courseId);
+
+        User student = userService.getUser();
+        Course course = courseService.findOne(courseId);
+
+        List<Exercise> exercises = exerciseService.findAllExercisesByCourseId(course, student);
+
+        for (Exercise exercise : exercises) {
+            List<Participation> participations = participationService.findByExerciseIdAndStudentIdWithEagerResults(exercise.getId(), student.getId());
+
+            Hibernate.initialize(exercise.getParticipations());
```

Comment:
I don't understand this line. In the next line your basically remove the participations, so why do you initialize it?

Reply:
i had a hibernate error. So i intitialize the participations and then replace it with the filtered ones. if there is a better solution please let me know

Reply:
just remove line 421, you set the participations to an empty HashSet in line 424 anyway, so there is not need to initialize them